### ISSUE TITLE
UpdateOne with array make array encapsulated inside an array

### ISSUE THREAD

**Issue by Poyoman39** [Automattic/mongoose#11452]
To my opinion this is a kind of critical bug, i keep available if you need more informations about it

**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
MyModel.updateOne({ _id: 'someId' }, { arrayPath: ['some', 'value'] }).exec()

=> i get in DB : { _id: 'someId', arrayPath: [ ['some', 'value'] ] } instead of { _id: 'someId', arrayPath: ['some', 'value'] }

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

This happens with mongoose 6.2.3, and as far as i know i didn't have this bug in 6.0.14
MongoDB is 5.0.6 CE

**Comment by Poyoman39** [comment]
I investigated a bit, and i suspect this commit https://github.com/Automattic/mongoose/commit/875ada1529081d67098f24ccc117bebf971a83ea to be the source of the bug.

**Comment by Poyoman39** [comment]
I can confirm this is working fine with 6.1.10

**Comment by AbdelrahmanHafez** [comment]
Welcome @Poyoman39 👋

The following script passes in mongoose 6.2.3, can you modify it to demonstrate the issue you're facing?
### 11452.js
```js
'use strict';
import mongoose from 'mongoose';
const { Schema } = mongoose;
import assert from 'node:assert';


run().catch(console.error);

async function run() {
  await mongoose.connect('mongodb://localhost:27017/test');

  await mongoose.connection.dropDatabase();

  const userSchema = new Schema({
    hobbies: [String],
    favoriteGames: [Schema.Types.Mixed]
  });

  const User = mongoose.model('User', userSchema);
  await User.create({
    hobbies: ['sports', 'cooking'],
    favoriteGames: ['Valorant', 'Among Us']
  });
  await User.updateOne(
    {},
    {
      hobbies: ['coding', 'swimming'],
      favoriteGames: ['League of Legends', 'Dota 2']
    }
  );

  const user = await User.findOne().lean();

  assert.deepStrictEqual(user.hobbies, ['coding', 'swimming']);
  assert.deepStrictEqual(user.favoriteGames, ['League of Legends', 'Dota 2']);
  console.log(mongoose.version);
  console.log('All assertions passed.');
}
```

### Output
```
6.2.3
All assertions passed.
```

**Comment by Poyoman39** [comment]
thank you @AbdelrahmanHafez , i found a way to reproduce my bug, and this is basically a problem on my schemas. However, there is a "breaking" change introduced with 6.2.1 i think

Please, try running this (i changed the `hobbies: [{ type: [String] }],` part :

```
import mongoose from 'mongoose';
import assert from 'node:assert';

const { Schema } = mongoose;

async function run() {
  await mongoose.connect('mongodb://localhost:27017/test');

  await mongoose.connection.dropDatabase();

  const userSchema = new Schema({
    // hobbies: [String],
    hobbies: [{ type: [String] }],
    favoriteGames: [Schema.Types.Mixed],
  });

  const User = mongoose.model('User', userSchema);
  await User.create({
    hobbies: ['sports', 'cooking'],
    favoriteGames: ['Valorant', 'Among Us'],
  });
  await User.updateOne(
    {},
    {
      hobbies: ['coding', 'swimming'],
      favoriteGames: ['League of Legends', 'Dota 2'],
    },
  );

  const user = await User.findOne().lean();

  console.log(mongoose.version);
  assert.deepStrictEqual(user.hobbies, ['coding', 'swimming']);
  assert.deepStrictEqual(user.favoriteGames, ['League of Legends', 'Dota 2']);
  console.log('All assertions passed.');
}

run().catch(console.error);

```

With 6.2.3 i get "AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:"
With 6.1.10 i get "All assertions passed."

So basically this is more a bug at validation step i think ?

**Comment by AbdelrahmanHafez** [comment]
It seems like this was a happy accident that fixed a bug.

The schema in your example means the path is an array of arrays:
`[{ type: [String] }]` is the equivalent of `[[String]]`.
If you're trying to achieve an array of strings you can simply go for: `[String]`.

Would that work?

**Comment by Poyoman39** [comment]
Hi @AbdelrahmanHafez sure this work, and i'm happy i have discovered this bug thanks to you and this happy accident ;)

However do you think this is a desirable behavior, that update with and ill-formatted value does not throw error, like if validation did not happen ? And/or take the assumption the given array should be pushed into an array ?

To me this look like hidden errors ... But i don't know if it's the default behavior in mongoose to transform update with element in push operations, when schema is an array

**Comment by Poyoman39** [comment]
Thank you for the precision :) Effectively i find it really usefull for string/number/ObjectId stuff or things like that.

**Comment by AbdelrahmanHafez** [comment]
Happy to help, and welcome to mongoose!

**Comment by vkarpov15** [comment]
I would say this is expected behavior. The schema definition does define a nested array, and, in this case, we have no way of telling that the dev meant to define an array of strings rather than an array of array of strings. Thanks for helping out with this @AbdelrahmanHafez

**Comment by AbdelrahmanHafez** [comment]
Happy to help 👍
